### ISSUE TITLE
Allow iceberg to use partitions created with a transformation functions.

### ISSUE THREAD

**Issue by dmgcodevil** [trinodb/trino#5825]
Problem:

we have a partition created using `day` transformation function:

```
  Types.NestedField.optional(0, FRONTDOOR_TIMESTAMP, Types.TimestampType.withZone())
   PartitionSpec.builderFor(....)
        .day(FRONTDOOR_TIMESTAMP)
        .build()
```

when query by `frontdoor_timestamp` , i.e.:

```
select count(1) from ... where frontdoor_timestamp > TIMESTAMP '2020-10-20 00:00:00.314 UTC';
```

it performs a full scan, in other words, it doesn't use that partition.

the problem in this line: https://github.com/prestosql/presto/blob/c1fd650b26017772550570fd1f3090591396e744/presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java#L668

this change was introduced in this [commit](https://github.com/prestosql/presto/commit/217ae3bd51d694983182a73daa1c3ea4f1b04761)

Once we removed this check iceberg started using partitions.

Second problem:

```
select count(1) from ... where frontdoor_timestamp > TIMESTAMP '2020-10-20 00:00:00.314 UTC';
```

fails: `Query 20201103_191606_00015_dwzq2 failed: Cannot create expression literal from io.prestosql.spi.type.LongTimestampWithTimeZone: io.prestosql.spi.type.LongTimestampWithTimeZone@26785952`

this issue was resolved by adding a special case in [getIcebergLiteralValue](https://github.com/prestosql/presto/blob/c1fd650b26017772550570fd1f3090591396e744/presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/ExpressionConverter.java#L169)

Will create a PR shortly

**Comment by phd3** [comment]
@dmgcodevil can you take a look at the PR https://github.com/prestosql/presto/pull/4932 before working on this one?
I think that should cover both these issues.

**Comment by dmgcodevil** [comment]
@phd3 looks good, also did some testing on real data. Do you have any ETA when PR will be merged?

**Comment by phd3** [comment]
@dmgcodevil Release 346 is going out likely today, so we should be able to merge it after that. (and then it'll be available in the next release)

**Comment by hashhar** [comment]
@phd3 @dmgcodevil Can this be closed? I can see that #4932 addresses this and was merged.
